### PR TITLE
fix: avoid duplicate auto-generated id

### LINKED ISSUE #2950: bug: Admin API concurrently created Router ID duplicate
### Issue description
Admin 接口并发创建 Router 得到的 ID 重复

### Environment

* apisix version (cmd: `apisix version`): 2.0
* OS: linux

### Minimal test code / Steps to reproduce the issue
1. 
2.
3.

### PR DESCRIPTION
**Pull request by spacewander**
Fix #2950

Signed-off-by: spacewander <spacewanderlzx@gmail.com>

### Pre-submission checklist:

* [ ] Did you explain what problem does this PR solve? Or what new features have been added?
* [ ] Have you added corresponding test cases?
* [ ] Have you modified the corresponding document?
* [x] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix/tree/master#community) first**

### CODE REVIEW

File: apisix/core/etcd.lua [hunk 1]
```diff
@@ -269,12 +269,15 @@ end
...
+    local etcd_cli, _, err = new()
     if not etcd_cli then
         return nil, err
     end
 
-    local res, err = etcd_cli:readdir(prefix .. key)
+    -- It will be better if we use snowflake algorithm like manager-api,
+    -- but we haven't found a good library. It costs too much to write
+    -- our own one as the admin-api will be replaced by manager-api finally.
+    local res, err = set("/gen_id", 1)
```
Comment (tokers) [inline apisix/core/etcd.lua:280]: I think we should add more comments here to explain why we need extra write operation here, it's not easy to understand if readers are out of the context of #2950 .

**Comment by membphis** [comment]
@idbeta @AloneFire you can make a try with this PR

File: apisix/core/etcd.lua (hunk 1 above)
Comment (spacewander) [reply apisix/core/etcd.lua:281]: Updated.

**Review by tokers — APPROVED** [review]

**Comment by idbeta** [comment]
> @idbeta @AloneFire you can make a try with this PR

This commit has solved this problem #2950

**Review by membphis — APPROVED** [review]

**Pull request merged**